### PR TITLE
docs(identity): document resource authorizations dependency on a database

### PR DESCRIPTION
**Pull request by Ben-Sheppard**
Related to https://github.com/camunda-cloud/identity/issues/1760

## What is the purpose of the change
We should document the requirements of the resource authorizations feature in Identity, which at the time of writing, is a configured database (Postgres currently supported)

## Are there related marketing activities
Nope

## When should this change go live?
The feature is already live on 8.2 and to avoid customer pain points it would be good to get the change out sooner rather than later, but I don't see it as a critical release.

## PR Checklist

- [x] My changes apply to an already released version, and I have added them to the relevant `/versioned_docs` directory, or they are not for an already released version.
- [x] My changes apply to future versions, and I have added them to the main `/docs` directory, or they are not for future versions.
- [x] My changes require an [Engineering review](https://github.com/camunda/camunda-platform-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned an engineering manager or tech lead as a reviewer, or my changes do not require an Engineering review.
- [x] My changes require a [technical writer review](https://github.com/camunda/camunda-platform-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @christinaausley as a reviewer, or my changes do not require a technical writer review.

### CODE REVIEW

**Review by christinaausley — DISMISSED** [review]
One non-blocking comment, and technical review in latest commit 👍

File: docs/self-managed/identity/deployment/configuration-variables.md [hunk 1]
```diff
@@ -44,10 +44,25 @@ and `WEBMODELER`.
 For the `WEBMODELER` value, only the `KEYCLOAK_INIT_<COMPONENT>_ROOT_URL` variable is required to be set.
 :::
 
+### Database configuration
+
```
[the remark this answers is not in the document]
Comment (christinaausley) [reply docs/self-managed/identity/deployment/configuration-variables.md:48]: I agree here. I understand the consistency perspective, but even in API documentation you can remove unused columns like default values. Good with removing this column and adding an admonition 👍 

Will keep an eye out to see if users are confused about this, but I think we're making it pretty clear here. Let me know if I can assist!

[the remark this answers is not in the document]
Comment (Ben-Sheppard) [reply docs/self-managed/identity/deployment/configuration-variables.md:48]: Sorry for the delay @christinaausley - I've made the changes now :)

**Review by christinaausley — APPROVED** [review]
Thank you @Ben-Sheppard !

**Pull request merged**